client/trino-jdbc/src/main/java/io/trino/jdbc/AbstractTrinoResultSet.java

```diff
@@ -92,8 +92,8 @@
 {
     private static final Pattern DATETIME_PATTERN = Pattern.compile("" +
             "(?<year>[-+]?\\d{4,})-(?<month>\\d{1,2})-(?<day>\\d{1,2})" +
-            "(?: (?<hour>\\d{1,2}):(?<minute>\\d{1,2})(?::(?<second>\\d{1,2})(?:\\.(?<fraction>\\d+))?)?)?" +
-            "\\s*(?<timezone>.+)?");
+            "( (?:(?<hour>\\d{1,2}):(?<minute>\\d{1,2})(?::(?<second>\\d{1,2})(?:\\.(?<fraction>\\d+))?)?)?" +
+            "(?:\\s*(?<timezone>.+))?)?");
```

Comment:
Why `(?: ... )?` around this whole construct? Instead of that, make the timezone block optional:

```
"\\s*(?<timezone>.+)?)?");
```

Reply:
Hi, @martint  I have made the change. Thanks for suggesting it. Can you take another look?